contracts/apps/AppProxyBase.sol

```diff
@@ -32,6 +33,20 @@ contract AppProxyBase is IAppProxy, AppStorage, DelegateProxy, KernelConstants {
         }
     }
 
+    /**
+     * @dev ERC897, whether it is a forwarding (1) or an upgradeable (2) proxy
+     */
+    function proxyType() public pure returns (uint256 proxyTypeId) {
+        return isUpgradeable() ? UPGRADEABLE : FORWARDING;
+    }
+
+    /**
+     * @dev ERC897, the address the proxy would delegate calls to
+     */
+    function implementation() public view returns (address) {
```

Comment:
We could save having one extra function if `APPinned` and `APUpgradeable` just implement `implementation()` rather than `getCode()` directly!

Reply:
Makes sense. We can actually do the same with `proxyType` and even remove `IAppProxy` interface, as it's essentially the same as the ERC 897 one.

Reply:
> We can actually do the same with proxyType and even remove IAppProxy interface, as it's essentially the same as the ERC 897 one.

👌 very nice!